operator/src/main/java/com/instaclustr/cassandra/operator/service/CassandraHealthCheckService.java

```diff
@@ -69,8 +71,13 @@ protected void runOneIteration() throws Exception {
         logger.debug("Checking health of cassandra instances...");
 
         for (final Map.Entry<DataCenterKey, DataCenter> cacheEntry : dataCenterCache.asMap().entrySet()) {
+        	
+        	logger.debug("Reconciling DataCenters...");
+        	
+        	controllerService.reconcileDataCenter(cacheEntry.getKey());
```

Comment:
Not sure if its the best idea to have the health check service, which just verifies the health of the Cassandra nodes, call out to the `ControllerService`.

Reply:
Created a seperate Repair service to repair missing components